File: Silver/2018.cpp
#include <bits/stdc++.h>

using namespace std;

int main(){
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    vector<int> v;
    int n;
    int s = 1, e = 1, sum = 0, cnt = 0;

    cin >> n;

    while(e <= n + 1) {
        if(sum > n){
            sum -= s++;
        } else if(sum < n) {
            sum += e++;
        } else if(sum == n) {
            cnt++;
            sum += e++;
        }
    }

    cout << cnt << "\n";

    return 0;
}

File: Silver/2018.py
n = int(input())
cnt, sum, start, end = 0, 1, 1, 1

while end <= n:
  if sum < n:
    end += 1
    sum += end
  elif sum > n:
    sum -= start
    start += 1
  else:
    cnt += 1
    end += 1
    sum += end
    
    
print(cnt)